app/controllers/npq_separation/migration/migrations_controller.rb

```diff
@@ -14,6 +14,20 @@ def create
     redirect_to npq_separation_migration_migrations_path
   end
 
+  def download_report
+    data_migration = Migration::DataMigration.find(params[:id])
+    failures = Migration::FailureManager.new(data_migration:).all_failures
+
+    failures_csv = CSV.generate(headers: true) do |csv|
```

Comment:
If we're changing this to a `CSV` it would be better in `FailureManager#all_failures_csv` so we can test it explicitly